xtask/src/build_docker_image.rs

```diff
@@ -321,6 +332,51 @@ pub fn build_docker_image(
     Ok(())
 }
 
+fn is_armv6() -> bool {
+    cfg!(any(
+        cross_triple = "arm-unknown-linux-gnueabi",
+        cross_triple = "arm-unknown-linux-musleabi"
+    ))
+}
+
+fn is_armv7() -> bool {
+    cfg!(any(
+        cross_triple = "armv7-unknown-linux-gnueabihf",
+        cross_triple = "armv7-unknown-linux-musleabihf"
+    ))
+}
+
+fn is_native(
+    platform: &str,
+    target: &crate::ImageTarget,
+    msg_info: &mut MessageInfo,
+) -> cross::Result<bool> {
+    Ok(match target.sub {
+        Some(_) => false,
+        None => match (platform, target.name.as_str()) {
+            ("linux/386", "i686-unknown-linux-gnu")
+            | ("linux/amd64", "x86_64-unknown-linux-gnu")
+            | ("linux/arm64" | "linux/arm64/v8", "aarch64-unknown-linux-gnu")
+            | ("linux/ppc64le", "powerpc64le-unknown-linux-gnu")
+            | ("linux/riscv64", "riscv64gc-unknown-linux-gnu")
+            | ("linux/s390x", "s390x-unknown-linux-gnu") => true,
+            ("linux/arm/v6", "arm-unknown-linux-gnueabi") if is_armv6() => {
+                note_host_target_detection(msg_info)?;
+                true
+            }
+            ("linux/arm" | "linux/arm/v7", "armv7-unknown-linux-gnueabihf") if is_armv7() => {
```

Comment:
this is already partly implemented with `ImagePlatform.target`

Reply:
We might have an issue in `Architecture::from_target` since it doesn't differentiate ARMv6 and ARMv7 like this does, and Docker platforms support both ARMv6 and ARMv7.